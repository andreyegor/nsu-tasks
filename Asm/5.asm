.include "things.asm"

.text
.globl main
main:
    call read_dec
	mv s0 a0#first number
	call read_dec
	mv s1 a0#second number
    li s2 0x80000000#iterator
    li s3 31#iterator too
    li s4 0xf0000000#out of range
    li a0 0 #result

    li s5 0
    and t0 s0 s4
    and t1 s1 s4
    xor s5 t0 t1 #sign

	beq t0 zero _main_second_sign
	sub s0 zero s0
_main_second_sign:
	beq t1 zero _main_loop
	sub s1 zero s1

_main_loop:
    srli s2 s2 1
    addi s3 s3 -1
    beq s2 zero _main_continue 


    and t0 s2 s1
    beq t0 zero _main_loop

    sll t1 s0 s3

    srl t2 t1 s3
    bne s0 t2 _main_err
    add a0 a0 t1

    and t2 a0 s4
    bne t2 zero _main_err

    j _main_loop
_main_continue:
	beq s5 zero _main_end
	sub a0 zero a0
_main_end:
	call write_dec
    exit 0
_main_err:
    exit 1