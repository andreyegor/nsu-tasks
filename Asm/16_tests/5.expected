















a
a
ac
aliquam
aliquam
aliquet
aliquet
arcu vitae
at
at
at
auctor
bibendum
condimentum
condimentum
condimentum
curabitur
cursus
diam
dignissim
donec
duis
egestas
eget
eget
eleifend
eleifend
elementum
enim
et
et
et
eu
fames
fermentum
habitant
iaculis
id
id
id
id
in
integer
justo
lectus
lorem
magna
malesuada
malesuada
mattis
metus
mi
morbi
nec sagittis aliquam 
netus
nibh
nibh
nisl
non
nulla
nulla
nunc
nunc
orci
pellentesque
pellentesque
phasellus
phasellusfaucibus
posuere
pretium
proin
purus
risus
sagittis
sapien
sapien
scelerisque
scelerisque
sed
sed
sed
semper
senectus
sodales ut eu sem
sollicitudin
tellus
tempus
tortor
tortor
tristique
tristique
turpis
ultrices
ultricies
urna
urna
urna
velit
venenatis
vestibulum
vitae
vitae
vitae
vulputate